src/stateProviders/provideLoggedInUser.js

```diff
@@ -19,33 +18,19 @@ export default provideState({
     initialize: effects => state => {
       const { setToken, setUser } = effects;
       const jwt = localStorage.getItem('EGO_JWT');
-      if (jwt) {
-        addHeaders({ authorization: `Bearer ${jwt}` });
-
-        try {
-          const gapi = global.gapi;
-          gapi.load('auth2', () => {
-            gapi.auth2.init({
-              client_id: googleAppId,
-            });
-          });
-        } catch (e) {
-          global.log(e);
-        }
+      if (validateJWT({ jwt })) {
         handleJWT({ jwt, setToken, setUser });
-
         // Get all integration keys from local storage
         SERVICES.forEach(service => {
           const storedToken = localStorage.getItem(`integration_${service}`);
           if (storedToken) {
             state.integrationTokens[service] = storedToken;
           }
         });
+        return { ...state, isLoadingUser: true };
       }
-      return {
-        ...state,
-        isLoadingUser: !!jwt,
-      };
+      localStorage.removeItem('EGO_JWT');
```

Comment:
Small thing, maybe use `effects.setToken(null)` here? It also clears out the ajax headers and all that jazz :)

Reply:
@hlminh2000 - yeah for sure, good call :)